### ISSUE TITLE
[Hot Reload] Hot reload currently fails if you have service scope validation enabled

### ISSUE THREAD

**Issue by PureWeen** [dotnet/maui#21081]
### Description

Run the MAUI samples project from our repository with hot reload enabled and you'll get an exception on the Dispatcher.

**Comment by drasticactions** [comment]
For context:

The MAUI Xaml Hot Reload agent calls for an IDispatcher to do UI activity

```csharp
        public static IDispatcher? GetDispatcher(this IElementHandler? handler)
        {
            return handler?.MauiContext?.Services.GetService(typeof(IDispatcher)) as IDispatcher;
        }
```

This breaks with Scope Validation enabled, since we're directly calling for the service from the root container. This code only runs with our agent, which is injected in Debug mode only. The sample enables scope validation, which then breaks the agent. It's been this way for two years and AFAIK we've never gotten feedback about it before, so I'm not sure how often people set that value, or if they only do it in Release mode builds. Regardless, we'll need a way to get a dispatcher (or do other ways of getting to the UI Thread)

**Comment by drasticactions** [comment]
This was fixed in VS Main. Once it lands in a public release of VS we can close this issue.

**Comment by jeremy-visionaid** [comment]
I find this happens with hot reload disabled also.. Setting `ValidateScopes = false` does avoid the crash though

**Comment by PureWeen** [comment]
This should be fixed in the latest release of VS 17.10 Preview
